samples/ADC/Program.cs

```diff
@@ -14,37 +13,37 @@ public class Program
+            AdcController adc = AdcController.GetDefault();
 
-            Debug.WriteLine("devs=" + devs);
+            // Minimum value is usually 0, but maximum depends on the board and the actual physical analog-to-digital converter.
+            // For example, STM32F769 has 12bit ADC, so the maximum value is 4095.
+            int maximum = adc.MaxValue;
+            int minimum = adc.MinValue;
+            Debug.WriteLine($"ADC minimum value is {minimum}, maximum - {maximum}");
 
-            AdcController adc1 = AdcController.GetDefault();
+            // Analog channels in nanoFramework are defined sequentially: 0, 1, 2, etc. However, these number are mapped inside firmware,
+            // and it is not obvious of how manufacturers datasheet corresponds to the nanoFramework channel numbers.
+            // It is different for every target board. For example, Channel 0 for STM32F769DISCOVERY board corresponds to ADC1_IN6,
+            // which is mapped to PA6 MCU pin, which is then routed to A0 pin on Arduino headers.
+            // Try looking for such information in reference target documentation pages: https://docs.nanoframework.net/content/reference-targets/index.html
 
-            int max1 = adc1.MaxValue;
-            int min1 = adc1.MinValue;
+            // For this sample, let's just open Channel 0. There's a good chance it is present on most boards.
+            AdcChannel channel0 = adc.OpenChannel(0);
 
-            Debug.WriteLine("min1=" + min1.ToString() + " max1=" + max1.ToString());
-
-            AdcChannel ac0 = adc1.OpenChannel(0);
-
-            // the following indexes are valid for STM32F769I-DISCO board
-            AdcChannel vref = adc1.OpenChannel(0);
-            AdcChannel vbat = adc1.OpenChannel(8);
-
-            // VP 
-            //AdcChannel ac3 = adc1.OpenChannel(3);
-            // VN 
             while (true)
             {
-                int value = ac0.ReadValue();
-                int valueVref = vref.ReadValue();
-                int valueVbat = vbat.ReadValue();
+                // Reading absolute value. 
+                int value = channel0.ReadValue();
 
-                double percent = ac0.ReadRatio();
+                // Reading ratio is may also be useful. You don't have to deal with ADC bits then.
+                double ratio = channel0.ReadRatio();
 
-                Debug.WriteLine("value0=" + value.ToString() + " ratio=" + percent.ToString());
-                Debug.WriteLine("verf" + valueVref.ToString() + " ratio=" + percent.ToString());
-                Debug.WriteLine("vbat" + valueVbat.ToString() + " ratio=" + percent.ToString());
+                // If left onconnected, analog pin will "float", so we'll read some random values.
+                // If you touch it with your finger, value will change.
+                // You may use a Dupoint wire to short it to GND or 3V3 pins,
```

Comment:
I get the point here, but this is not a  good advice... 
There are ADCs that are not 3.3V tolerant and connecting those to 3.3V has the potential to fry it. 
Worth mentioning too that touching an MCU pin that has no ESD protection can have downside effects on the part health.

Reply:
Will change the wording, although I'm yet to see a VCC 3V3 MCU that has 3V3 intolerant analog pins, even if VRef is less. VCC=1.8V chip are a different story, but I don't think nF targets any of such boards?..

Reply:
TI CC3220 and CC1352R

Reply:
I admit my defeat, these are the first one for me.

Reply:
Why a defeat!? 😯 You've mentioned that and I've pointed you to those...